src/idl_gen_general.cpp

```diff
@@ -731,6 +731,7 @@ void GenStruct(StructDef &struct_def, std::string *code_ptr) {
        it != struct_def.fields.vec.end();
        ++it) {
     auto &field = **it;
+    assert(field.value.type.base_type != BASE_TYPE_ARRAY);
```

Comment:
It's nicer to handle errors in a centralized location, so I'm afraid we're going to have to pass an error back. To make that simpler you can stick a `std::string error_` in `BaseGenerator` that you then check for not being empty in `flatc.cpp`.  And return early from this function.